store/store_download.go

```diff
@@ -298,15 +410,17 @@ func downloadImpl(ctx context.Context, name, sha3_384, downloadURL string, user
 			}
 		}
 
-		if cancelled(ctx) {
-			return fmt.Errorf("The download has been cancelled: %s", ctx.Err())
+		if cancelled(downloadCtx) {
+			return fmt.Errorf("The download has been cancelled: %s", downloadCtx.Err())
 		}
 		var resp *http.Response
 		cli := s.newHTTPClient(nil)
-		resp, finalErr = s.doRequest(ctx, cli, reqOptions, user)
-
-		if cancelled(ctx) {
-			return fmt.Errorf("The download has been cancelled: %s", ctx.Err())
+		resp, finalErr = s.doRequest(downloadCtx, cli, reqOptions, user)
+		if cancelled(downloadCtx) {
+			if finalErr == nil {
+				resp.Body.Close()
```

Comment:
this if is new and also not exercised,  do we expect this to happen  now?

Reply:
I haven't seen this happening, just noticed a potential problem - per client.Do() doc:
"If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close."

I'm not sure how to to test it and it's not strictly related to the bug being fixed there, so I've removed this change for now and will leave it for a separate followup PR.